MonstieWash/Assets/Scripts/UIManager.cs

```diff
@@ -1,28 +1,42 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using TMPro;
 using UnityEngine;
 using UnityEngine.UI;
 
 public class UIManager : MonoBehaviour
 {
-    [SerializeField] private TextMeshProUGUI text;
+    public GameObject m_TaskContainer;
+    public GameObject m_TaskTextPrefab;
```

Comment:
These should simply be serialized private fields. They don't have any reason to be public. Also outward-facing fields don't require the prefix.
```suggestion
    [SerializeField] private GameObject taskContainer;
    [SerializeField] private GameObject taskTextPrefab;
```